util/short_client_token.py

```diff
@@ -1,103 +1,100 @@
 import base64
 import datetime
 import logging
+
 from jwt.algorithms import HMACAlgorithm
 
-class ShortClientTokenTool(object):
 
+class ShortClientTokenTool:
     ALGORITHM = 'HS256'
     signer = HMACAlgorithm(HMACAlgorithm.SHA256)
 
     @classmethod
     def adobe_base64_encode(cls, to_encode):
-        """A modified base64 encoding that avoids triggering an Adobe bug.
+        """
+        A modified base64 encoding that avoids triggering an Adobe bug.
 
-        The bug seems to happen when the 'password' portion of a
-        username/password pair contains a + character. So we replace +
-        with :. We also replace / (another "suspicious" character)
+        The bug seems to happen when the 'password' portion of a username/password pair contains
+        a + character. So we replace + with :. We also replace / (another "suspicious" character)
         with ;. and strip newlines.
         """
-        if isinstance(to_encode, unicode):
+        if isinstance(to_encode, str):
             to_encode = to_encode.encode("utf8")
-        encoded = base64.encodestring(to_encode)
+
+        encoded = base64.encodebytes(to_encode)
+
         return encoded.replace(b"+", b":").replace(b"/", b";").replace(b"=", b"@").strip()
 
     @classmethod
     def adobe_base64_decode(cls, to_decode):
         """Undoes adobe_base64_encode."""
-        if isinstance(to_decode, unicode):
-            to_decode = to_decode.encode("utf8")        
+        if isinstance(to_decode, str):
+            to_decode = to_decode.encode("utf8")
+
         to_decode = to_decode.replace(b":", b"+").replace(b";", b"/").replace(b"@", b"=")
-        return base64.decodestring(to_decode)
 
-    # The JWT spec takes January 1 1970 as the epoch.
-    JWT_EPOCH = datetime.datetime(1970, 1, 1)
+        return base64.decodebytes(to_decode)
+    
+    JWT_EPOCH = datetime.datetime(1970, 1, 1)   # The JWT spec takes January 1 1970 as the epoch.
```

Comment:
In circulation code we've changed all of our datetime handling so that our datetime objects are timezone-aware and UTC (rather than being timezone-naive). The fact that this appears to be local time is a problem, though it might be a problem we already have. Rather than just change this to a UTC timestamp (which might break the registry/CM relationship) let's investigate doing the timezone stuff in a follow-up branch.